deps/rabbit/src/rabbit_amqqueue.erl

```diff
@@ -652,13 +652,47 @@ priv_absent(QueueName, QPid, _IsDurable, alive) ->
             'ok' | rabbit_types:channel_exit() | rabbit_types:connection_exit().
 
 assert_equivalence(Q, DurableDeclare, AutoDeleteDeclare, Args1, Owner) ->
-    QName = amqqueue:get_name(Q),
-    DurableQ = amqqueue:is_durable(Q),
-    AutoDeleteQ = amqqueue:is_auto_delete(Q),
-    ok = check_exclusive_access(Q, Owner, strict),
-    ok = rabbit_misc:assert_field_equivalence(DurableQ, DurableDeclare, QName, durable),
-    ok = rabbit_misc:assert_field_equivalence(AutoDeleteQ, AutoDeleteDeclare, QName, auto_delete),
-    ok = assert_args_equivalence(Q, Args1).
+    case ignore_checks_on_qq_redeclaration(Args1, Q) of
+        true ->
+            ok;
+        _ ->
+            QName = amqqueue:get_name(Q),
+            DurableQ = amqqueue:is_durable(Q),
+            AutoDeleteQ = amqqueue:is_auto_delete(Q),
+            ok = check_exclusive_access(Q, Owner, strict),
+            ok = rabbit_misc:assert_field_equivalence(DurableQ, DurableDeclare, QName, durable),
+            ok = rabbit_misc:assert_field_equivalence(AutoDeleteQ, AutoDeleteDeclare, QName, auto_delete),
+            ok = assert_args_equivalence(Q, Args1)
+    end.
+
+
+ignore_checks_on_qq_redeclaration(New, Q) ->
```

Comment:
This function right now is both a predicate and performs certain checks. Maybe
its name should then be `perform_limited_equivalence_checks_on_qq_redeclaration`.

Reply:
Restructured the code a bit.